### ISSUE TITLE
Proposal: remove support for the node_syscall extension

### ISSUE THREAD

**Issue by nevkontakte** [gopherjs/gopherjs#1328]
The extension has been deprecated for about 3 years (since https://github.com/gopherjs/gopherjs/pull/1111), and has been disabled by default since then. So far I haven't heard of anyone asking to continue supporting it. This extension has known issues in the memory management department and doesn't support all types of arguments (e.g. https://github.com/gopherjs/gopherjs/issues/993). I am not even sure it builds under the modern node versions.

I would like to do the following:

 1. Move the source code into a separate repository under the gopherjs organization and mark it archived. This will keep sources available in case anyone is using it for some other purpose, and it would clearly indicate its status an unmaintained.
 2. Remove `legacy_syscall` build tag from the code base.
 3. Remove the option of using different GOOS/GOARCH for user code which was added as a compatibility option in #1111 and fix them to GOOS=js GOARCH=ecmascript.

Inviting @flimzy and anyone else to comment.

**Comment by flimzy** [comment]
I'm in principle in favor of the proposal. I'm only concerned that the fact that we haven't done a complete release in quite some time may mean that there may be those using it, who aren't aware yet of the changes.

As such, I wonder if for the next major release (when generics are complete) we could move it to its own repo first, which will make the deprecation status more obvious, and likely bring out of the woodwork anyone who really needs it.  Then assuming no major objections, do the rest of the removal for the following major release.

**Comment by nevkontakte** [comment]
Yeah, that's a fair argument. I'll admit I am a bit overeager to get rid of it because [debugging mystery sigfaults](https://github.com/gopherjs/gopherjs/pull/1329/commits/b579c8f75db49a5668877659ef34e1a72c5df798) was not fun 😅 But making a full release before deprecating it entirely does seem like a responsible way of doing things.
